### PR TITLE
Remove EmailPreviewTemplate

### PR DESCRIPTION
**Pull request by quis**
The only thing using it was the admin app and we moved the code into notifications-admin in https://github.com/alphagov/notifications-admin/pull/5266

### CODE REVIEW

File: tests/test_template_types.py [hunk 1]
```diff
@@ -356,7 +355,6 @@ def test_markdown_in_templates(
     "template_class, template_type, extra_attributes",
     [
         (HTMLEmailTemplate, "email", 'style="word-wrap: break-word; color: #1D70B8;"'),
-        (EmailPreviewTemplate, "email", 'style="word-wrap: break-word; color: #1D70B8;"'),
```
Comment (quis) [inline tests/test_template_types.py:359]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR797

File: tests/test_template_types.py [hunk 2]
```diff
@@ -940,7 +938,6 @@ def test_sms_templates_have_no_subject(template_class, template_json):
 
 def test_subject_line_gets_applied_to_correct_template_types():
     for cls in [
-        EmailPreviewTemplate,
```
Comment (quis) [inline tests/test_template_types.py:943]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR498-R507

File: tests/test_template_types.py [hunk 3]
```diff
@@ -957,7 +954,6 @@ def test_subject_line_gets_applied_to_correct_template_types():
 @pytest.mark.parametrize(
     "template_class, template_type, extra_args",
     (
-        (EmailPreviewTemplate, "email", {}),
```
Comment (quis) [inline tests/test_template_types.py:960]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR510-R532

File: tests/test_template_types.py [hunk 4]
```diff
@@ -974,7 +970,6 @@ def test_subject_line_gets_replaced(template_class, template_type, extra_args):
...
```
Comment (quis) [inline tests/test_template_types.py:977]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR815-R828

File: tests/test_template_types.py [hunk 5]
```diff
@@ -1289,16 +1274,6 @@ def test_is_message_empty_email_and_letter_templates_tries_not_to_count_chars(
...
-        (
-            EmailPreviewTemplate,
-            "email",
-            {"redact_missing_personalisation": True},
-            [
-                mock.call("content", {}, html="escape", markdown_lists=True, redact_missing_personalisation=True),
-                mock.call("subject", {}, html="escape", redact_missing_personalisation=True),
-                mock.call("((email address))", {}, with_brackets=False),
-            ],
-        ),
```
Comment (quis) [inline tests/test_template_types.py:1301]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR535-R583

File: tests/test_template_types.py [hunk 6]
```diff
@@ -1548,7 +1495,6 @@ def test_templates_make_quotes_smart_and_dashes_en(
     (
         HTMLEmailTemplate,
         PlainTextEmailTemplate,
-        EmailPreviewTemplate,
```
Comment (quis) [inline tests/test_template_types.py:1551]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR883-R899

File: tests/test_template_types.py [hunk 7]
```diff
@@ -1613,12 +1559,6 @@ def test_smart_quotes_removed_from_long_template_in_under_a_second():
             ),
             ["subject", "content"],
         ),
-        (
-            EmailPreviewTemplate(
-                {"content": "((content))", "subject": "((subject))", "template_type": "email"},
-            ),
-            ["subject", "content"],
-        ),
```
Comment (quis) [inline tests/test_template_types.py:1621]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR603-R631

File: tests/test_template_types.py [hunk 8]
```diff
@@ -1647,62 +1587,6 @@ def test_html_template_can_inject_personalisation_with_special_characters():
...
-        {"from_name": "Example service"},
-        pytest.param({}, marks=pytest.mark.xfail),
-    ],
-)
-def test_email_preview_shows_from_name(extra_args):
-    template = EmailPreviewTemplate(
-        {"content": "content", "subject": "subject", "template_type": "email"}, **extra_args
-    )
-    assert '<th scope="row">From</th>' in str(template)
-    assert "Example service" in str(template)
```
Comment (quis) [inline tests/test_template_types.py:1662]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR902-R914

File: tests/test_template_types.py [hunk 9]
```diff
@@ -1647,62 +1587,6 @@ def test_html_template_can_inject_personalisation_with_special_characters():
...
-
-
-def test_email_preview_escapes_html_in_from_name():
-    template = EmailPreviewTemplate(
-        {"content": "content", "subject": "subject", "template_type": "email"}, from_name='<script>alert("")</script>'
-    )
-    assert "<script>" not in str(template)
-    assert '&lt;script&gt;alert("")&lt;/script&gt;' in str(template)
```
Comment (quis) [inline tests/test_template_types.py:1670]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR917-R922

File: tests/test_template_types.py [hunk 10]
```diff
@@ -1647,62 +1587,6 @@ def test_html_template_can_inject_personalisation_with_special_characters():
...
-        {"reply_to": "test@example.com"},
-        pytest.param({}, marks=pytest.mark.xfail),
-    ],
-)
-def test_email_preview_shows_reply_to_address(extra_args):
-    template = EmailPreviewTemplate(
-        {"content": "content", "subject": "subject", "template_type": "email"}, **extra_args
-    )
-    assert '<th scope="row">Reply&nbsp;to</th>' in str(template)
-    assert "test@example.com" in str(template)
```
Comment (quis) [inline tests/test_template_types.py:1685]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR925-R941

File: tests/test_template_types.py [hunk 11]
```diff
@@ -1647,62 +1587,6 @@ def test_html_template_can_inject_personalisation_with_special_characters():
...
-)
-def test_email_preview_shows_recipient_address(
-    template_values,
-    expected_content,
-):
-    template = EmailPreviewTemplate(
-        {"content": "content", "subject": "subject", "template_type": "email"},
-        template_values,
-    )
-    assert expected_content in str(template)
```
Comment (quis) [inline tests/test_template_types.py:1703]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR944-R959

File: tests/test_template_types.py [hunk 12]
```diff
@@ -1884,7 +1768,6 @@ def test_message_is_not_too_long_ignoring_prefix(template_class):
 @pytest.mark.parametrize(
     "template_class, template_type, kwargs",
     [
-        (EmailPreviewTemplate, "email", {}),
```
Comment (quis) [inline tests/test_template_types.py:1887]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR976-R979 (with number just under limit 👍🏻 )

File: tests/test_template_types.py [hunk 13]
```diff
@@ -1899,7 +1782,6 @@ def test_message_too_long_limit_bigger_or_nonexistent_for_non_sms_templates(temp
...
```
Comment (quis) [inline tests/test_template_types.py:1902]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR962-R966

File: tests/test_template_types.py [hunk 14]
```diff
@@ -1914,7 +1796,6 @@ def test_content_size_in_bytes_for_email_messages(template_class, template_type,
...
```
Comment (quis) [inline tests/test_template_types.py:1917]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR969-R973

File: tests/test_template_types.py [hunk 15]
```diff
@@ -1986,7 +1866,6 @@ def test_multiple_newlines_in_letters(
     [
         (PlainTextEmailTemplate, "email", {}),
         (HTMLEmailTemplate, "email", {}),
-        (EmailPreviewTemplate, "email", {}),
```
Comment (quis) [inline tests/test_template_types.py:1989]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR982-R996

File: tests/test_template_types.py [hunk 16]
```diff
@@ -2000,7 +1879,6 @@ def test_whitespace_in_subjects(template_class, template_type, subject, extra_ar
 @pytest.mark.parametrize(
     "template_class",
     [
-        EmailPreviewTemplate,
```
Comment (quis) [inline tests/test_template_types.py:2003]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR999-R1006

File: tests/test_template_types.py [hunk 17]
```diff
@@ -2356,45 +2234,6 @@ def test_text_messages_collapse_consecutive_whitespace(
...
-            (
-                '<p style="Margin: 0 0 20px 0; font-size: 19px; line-height: 25px; '
-                'color: #0B0C0C;">'
-                '<a style="word-wrap: break-word; color: #1D70B8;" '
-                "href=\"pre<span class='placeholder'>&#40;&#40;var&#41;&#41;</span>post\">"
-                "Example"
-                "</a>"
-                "</p>"
-            ),
-        ),
```
Comment (quis) [inline tests/test_template_types.py:2397]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR1009-R1051

File: tests/test_template_types.py [hunk 18]
```diff
@@ -2444,17 +2283,6 @@ def test_rendered_letter_template_for_print_can_toggle_notify_tag_and_always_hid
...
-            EmailPreviewTemplate,
-            (
-                '<hr style="border: 0; height: 1px; background: #B1B4B6; Margin: 30px 0 30px 0;">'
-                '<p style="Margin: 0 0 20px 0; font-size: 19px; line-height: 25px; color: #0B0C0C;">'
-                '<a style="word-wrap: break-word; color: #1D70B8;" href="https://www.example.com">'
-                "Unsubscribe from these emails"
-                "</a>"
-                "</p>\n"
-            ),
-        ),
```
Comment (quis) [inline tests/test_template_types.py:2457]: https://github.com/alphagov/notifications-admin/pull/5266/files#diff-e7949c604d3f8a5b6bc46af2528b92206f955af09f02fecb5086b004c00041abR1054-R1080

**Review by tombye — CHANGES_REQUESTED** [review]
This all looks good to go (thanks for the really helpful comments on the changes to the tests).

Think it needs a version bump and changelog entry, then good to go 👍🏻

**Comment by quis** [comment]
@tombye  Version bump and changelog done

**Review by tombye — APPROVED** [review]
Thanks for making the changes @quis. Now looks good to go 👍🏻

**Review by tombye — APPROVED** [review]
2nd approval as changes since created merge conflicts (now resolved) 👍🏻

**Pull request merged**